### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by TechnikEmpire**
- Adds Tensorboard logging to model directories.
 - Tensorboard logging include per-epoch confusion matrix generation.
 - Modifies scripts to include training for a large majority of all available architectures available on TFHub.
 - I've trained every kind of Mobilenet V2 and four efficientnets, MNasNet-A, Resnet 50 V2 and Inception V3 and created a whole section in the repo that shows current model stats. Models will be linked in this PR which should then be uploaded as part of a v 1.2.1 tagged release.
 - With Efficientnet B3, we've leaped past previous accuracies and hit almost 98% accuracy, which is no surprise since it's state of the art.
 - Several changes to the readme to bring things up to speed with the TF 2.* trainer, where models live now (releases page), etc.

### CODE REVIEW

**Comment by TechnikEmpire** [comment]
https://technikempire.com/models/efficientnet_b0.zip
https://technikempire.com/models/efficientnet_b1.zip
https://technikempire.com/models/efficientnet_b2.zip
https://technikempire.com/models/efficientnet_b3.zip
https://technikempire.com/models/inception_v3_224.zip
https://technikempire.com/models/mobilenet_v2_035_224.zip
https://technikempire.com/models/mobilenet_v2_050_224.zip
https://technikempire.com/models/mobilenet_v2_075_224.zip
https://technikempire.com/models/mobilenet_v2_100_224.zip
https://technikempire.com/models/mobilenet_v2_140_224.zip
https://technikempire.com/models/nasnet_a_224.zip
https://technikempire.com/models/resnet_v2_50_224.zip

Please don't hammer my server, this is just for @GantMan to pull them down and then upload them as attachments to the release. Some models may not be possible to host on the releases page, like Inception and Resnet which are both several hundred megs.

**Comment by peerwaya** [comment]
Great work @TechnikEmpire . I can see that for `saved_model.tflite` has increased, about 45mb making it quite large to add to a mobile app compared to about 24mb in the current release though this is for `efficientnet-b3`. Can integer post training quantization be done to further reduced the size?

**Comment by TechnikEmpire** [comment]
@peerwaya I'm not sure about quantizing the tflites. I'm fairly new to tensorflow, though I do vaguely recall seeing something about this in some tf code.

If you look at the model stats, the effnenet b0 isn't a whole lot better than the mobilenet v2 140 224. I believe this is simply because our accuracy ceiling is directly tied to training and validation quality, but anyway.

I personally would just stick with the best mobilenet. It's more than twice as fast for forward inference and you're only losing a bit of accuracy. 

Back to my point about accuracy ceiling, I heavily modified the original training set and even entirely replaced the hentai category. That sole change jumped us from about 93 percent on validation to about 97. Just to further clarify where the big leap came from.

**Comment by TechnikEmpire** [comment]
Looks like it is doable with a bit or custom code. Perhaps this can become a TODO to build right in to the training tool. However I'm insanely busy and I think @GantMan is too so maybe someone can else can pick it up. I'll be out of the loop for quite some time.

https://www.tensorflow.org/lite/performance/post_training_quantization

**Comment by BerryRB** [comment]
@TechnikEmpire Hi, when training these models, does the training data include all kinds object categories or only images of normal people? If all kinds, how many categories  for training neutral images? And what is the proportion of neutral  people?  Look forward to your reply！
